rpc/handlers_test.go

```diff
@@ -368,6 +396,27 @@ func TestBlockWithTxs(t *testing.T) {
 
 	t.Run("blockID - number", func(t *testing.T) {
 		mockReader.EXPECT().BlockByNumber(latestBlockNumber).Return(latestBlock, nil).Times(2)
+		mockReader.EXPECT().L1Head().Return(nil, db.ErrKeyNotFound).Times(2)
+
+		blockWithTxHashes, rpcErr := handler.BlockWithTxHashes(rpc.BlockID{Number: latestBlockNumber})
+		require.Nil(t, rpcErr)
+
+		blockWithTxs, rpcErr := handler.BlockWithTxs(rpc.BlockID{Number: latestBlockNumber})
+		require.Nil(t, rpcErr)
+
+		assert.Equal(t, blockWithTxHashes.BlockHeader, blockWithTxs.BlockHeader)
+		assert.Equal(t, len(blockWithTxHashes.TxnHashes), len(blockWithTxs.Transactions))
+
+		checkLatestBlock(t, blockWithTxHashes, blockWithTxs)
+	})
+
+	t.Run("blockID - number accepted on l1", func(t *testing.T) {
```

Comment:
Should we `assert.Equal(t, rpc.StatusAcceptedL1, block.Status)`, similar to the test for `BlockWithTxHashes`?

Reply:
The `checkLatestBlock` makes sure that the headers are equal for `blockWithTxHashes` and `blockWithTx` and since we check the Status in `BlockWithTxHasehes` we don't need to recheck it in `blockWithTx` explicitly. This was the convention we used for test other fields as well.